builds: add fedora mirror container

dvdhrm opened the pull request:
This adds a new dockerfile called `fedmir`. It takes as input a set of
fedora packages, an architecture, as well as a fedora release version,
and then downloads the specified packages and creates a fedora
repository. It then uses the official `httpd` image as base and serves
this manually created fedora image.

This container allows to create custom fedora mirrors with a selected
set of packages. The idea is to build these with all packages needed
for our tests, pushing it to `Github Packages`, and then using it in our
`Github Actions` infrastructure to avoid accessing resources from
outside the CI system.

dvdhrm commented:
I forgot to include: With this pushed to Github-Packages, we could simply include it as a `service container` in github-actions, and thus have all these packages locally available with no external internet access required anymore.

teg commented:
I think we may need to also set `module_platform_id` in case of cross version builds. I think it might work also without, but it is all a bit unclear.

dvdhrm commented:
> I think we may need to also set `module_platform_id` in case of cross version builds. I think it might work also without, but it is all a bit unclear.

Yeah, I tried to investigate a bit. Unclear what exactly it is used for, and it seems to work for me without. But apparently, it is required to resolve module dependencies. I now added it to the `dnf download` command, but I auto-generate it to `platform:f<RELEASE>`. Seems to work.

larskarlitski commented on builds/fedmir:
```diff
@@ -0,0 +1,83 @@
+# Import our mirror parameters.
+ARG     FEDMIR_ARCH
+ARG     FEDMIR_PACKAGES
+ARG     FEDMIR_RELEASE
```
These are documented as having default values `"x86_64"`, `"base emacs vim"`, and `"rawhide"`, but the default is missing here.

Note that `base` is not a valid package name. There's `basesystem`. Maybe we could just use whatever is in the `@core` comps group? (However, `dnf download` doesn't support downloading groups. I have a patch upcoming for that.)

larskarlitski commented on builds/fedmir:
```diff
@@ -0,0 +1,83 @@
+#
+
+# Import container arguments (must be before any `FROM`).
+ARG     FEDMIR_BUILD="docker.io/library/fedora:31"
```
This is documented as using `:latest`.

dvdhrm commented:
@larskarlitski I fixed the issues you mentioned. I also reworked this to use `dnf install --downloadonly`.

It is now merged in `master`. Feel free to adjust it if you still have comments!

Thanks a lot!

larskarlitski commented:
> Thanks a lot!

Thank you!
